### ISSUE TITLE
Use NSwag settings class

### ISSUE THREAD

**Issue by RicoSuter** [agc93/Cake.NSwag#1]
Hi, can't you use the NSwag settings classes (or inherit from them) so that all options are available?

**Comment by agc93** [comment]
Yeah, I thought about that when I was building but at the time it didn't really fit my design. I might revisit this. 

I particularly like the idea of inheriting to get all the existing settings without sacrificing customisation. I assume the NSwag settings aren't `sealed` or anything?

**Comment by agc93** [comment]
Ah, I remember the big reason I didn't at the time: my existing settings classes are for a given target, _regardless of source_, whereas the NSwag ones are for a specific source and target combination (i.e. `SwaggerToCSharpGeneratorSettings`).

I will continue having a look because I agree having all the settings available would be nice, but I also like the elegance of the same settings interface for any source...

**Comment by RicoSuter** [comment]
No, they aren't sealed... and yes, the big benefit is that you have always access to all settings and don't have to always extend your own settings class.. Btw. thanks for implementing this addin and using NSwag

**Comment by RicoSuter** [comment]
Why dont you provide a sub class of the original settings class as second parameter of the From\* and To\* methods? They can stll return the same unified type...

**Comment by agc93** [comment]
Hmmm, yeah, that could work! I've started a branch to assess going "big bang" into the original settings classes, but that seems simpler. I'll have to take a look.

**Comment by agc93** [comment]
I've just published a new `nswag-settings` feature branch that uses a sort of hybrid approach which also doesn't break existing usages. Basically, there's a new `WithSettings` mode that lets you use an NSwag settings object, as per below:

```
NSwag.FromSwaggerSpecification("./swagger.json")
    .ToTypeScriptClient("./client.ts", s =>
        s.WithSettings(new SwaggerToTypeScriptClientGeneratorSettings
        {
            Template = TypeScriptTemplate.Angular2,
            PromiseType = PromiseType.Promise
        }));
```

That way, we can keep the simplicity of the old one, doesn't break existing usages, but still can use operation-specific settings if necessary.

Happy to take any thoughts, but I will just do a bit more testing and will probably push it out as v0.2 next week.

**Comment by agc93** [comment]
Included in [0.2.0 release](https://github.com/agc93/Cake.NSwag/commit/a66333a1cacc75b96d448614cbfc6346466f3381).

Now available on the [NuGet Gallery](https://www.nuget.org/packages/Cake.NSwag/0.2.0).
